File: app/src/main/java/com/semye/android/retrofit/MainActivity.java
package com.semye.android.retrofit;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.semye.android.R;

import java.io.IOException;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        doGet();

    }

    private void doGet() {
        try {
            Retrofit retrofit = new Retrofit.Builder()
                    .baseUrl("https://www.baidu.com")
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
            ApiServices apiService = retrofit.create(ApiServices.class);
            Call<ResponseBody> service = apiService.getService("com", "222");
            service.enqueue(new Callback<ResponseBody>() {
                @Override
                public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
                    if (response.isSuccessful()) {
                        try {
                            System.out.println(response.body().string());
                        } catch (IOException e) {
                            e.printStackTrace();
                        }
                    } else {

                    }
                }

                @Override
                public void onFailure(Call<ResponseBody> call, Throwable t) {
                    System.out.println(t.getMessage());
                }
            });
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

}

File: app/src/main/java/com/semye/android/jetpack/viewmodel/MainActivity.kt
package com.semye.android.jetpack.viewmodel

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.semye.android.R

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main_viewmodel)
        val textView = findViewById<TextView>(R.id.text)
        val button = findViewById<Button>(R.id.button)

        val userViewModel = ViewModelProvider(this, ViewModelProvider.NewInstanceFactory()).get(
            UserViewModel::class.java
        )

        button.setOnClickListener { userViewModel.request() }

        userViewModel.user.observe(this, Observer {
            textView.text = it.name
        })
    }
}
